app/Services/ArticleService.php

```diff
@@ -37,6 +37,10 @@ public function getCategorizedArticlesByMagazine($magazineId)
         // For now, simply sorting categories alphabetically.
         ksort($categories);
 
+        foreach ($categories as $index => $category) {
+            $categories[$index] = collect($category)->sortByDesc('created_at');
+        }
```

Comment:
Lets discuss the best place for this. There are two views:

- Sorting is business logic and it must go in services, so its right place.
- We must minimise efforts and should do things using best tool. For sorting, database (order by) is best tool. Here we need to run a foreach loop again, which could be avoided by order by.

I **Personally prefer** second option because I feel it minimise efforts (more optimised) and best tool (database) is doing what is it made for. Any reason why you used option 1?